erpnext/healthcare/doctype/patient_encounter/patient_encounter.py

```diff
@@ -33,6 +34,100 @@ def set_title(self):
 		self.title = _('{0} with {1}').format(self.patient_name or self.patient,
 			self.practitioner_name or self.practitioner)[:100]
 
+	@frappe.whitelist()
+	@staticmethod
+	def get_applicable_treatment_plans(encounter):
+		patient = frappe.get_doc('Patient', encounter['patient'])
+
+		plans = None
+		plan_filters = {}
+		plan_filters['name'] = ['in', []]
+
+		age = patient.age
+		if age:
+			plan_filters['patient_age_from'] = ['>=', age.years]
+			plan_filters['patient_age_to'] = ['<=', age.years]
+
+		gender = patient.sex
+		if gender:
+			plan_filters['gender'] = gender
+
+		diagnosis = [_diagnosis['diagnosis'] for _diagnosis in encounter['diagnosis']]
+		if diagnosis:
+			filters = [
+				['diagnosis', 'in', diagnosis],
+				['parenttype', '=', 'Treatment Plan Template'],
+			]
+			diagnosis = frappe.get_list('Patient Encounter Diagnosis', filters=filters, fields='*')
+			plan_names = [_diagnosis['parent'] for _diagnosis in diagnosis]
+			plan_filters['name'][1].extend(plan_names)
+
+		symptoms = [symptom['complaint'] for symptom in encounter['symptoms']]
+		if symptoms:
+			filters = [
+				['complaint', 'in', symptoms],
+				['parenttype', '=', 'Treatment Plan Template'],
+			]
+			symptoms = frappe.get_list('Patient Encounter Symptom', filters=filters, fields='*')
+			plan_names = [symptom['parent'] for symptom in symptoms]
+			plan_filters['name'][1].extend(plan_names)
```

Comment:
If I don't enter any symptoms or diagnosis it breaks.
![image](https://user-images.githubusercontent.com/24353136/130764387-21bbda47-4360-484a-ac95-0d3ea12c4898.png)

You will need to check before running the list comprehension:

`if encounter.get('diagnosis')`:

Same for symptoms

Reply:
This seems to happen only on new encounters. For old encounters where symptoms/diagnosis is added and then removed, it is storing an empty list. Fixed it.